plugins/LayerView/layers.shader

```diff
@@ -3,33 +3,163 @@ vertex =
     uniform highp mat4 u_modelViewProjectionMatrix;
     uniform lowp float u_active_extruder;
     uniform lowp float u_shade_factor;
+    uniform highp int u_layer_view_type;
 
+    attribute highp float a_extruder;
+    attribute highp float a_line_type;
     attribute highp vec4 a_vertex;
     attribute lowp vec4 a_color;
+    attribute lowp vec4 a_material_color;
+
     varying lowp vec4 v_color;
+    varying float v_line_type;
+
     void main()
     {
         gl_Position = u_modelViewProjectionMatrix * a_vertex;
-        // shade the color depending on the extruder index stored in the alpha component of the color
-        v_color = (a_color.a == u_active_extruder) ? a_color : a_color * u_shade_factor;
-        v_color.a = 1.0;
+        v_color = a_color;
+        if ((a_line_type != 8.0) && (a_line_type != 9.0)) {
```

Comment:
a_line_type seems to be an enum. Could you also use it as such. I have no idea what 8.0 and 9.0 indicate right now.